src/Encoding.chpl

```diff
@@ -0,0 +1,254 @@
+/* Documentation for Encoding */
+module Encoding {
+
+  use IO;
+
+  // wireTypes
+  const varint = 0;
+  const lengthDelimited = 2;
+  const fixed64Type = 1;
+  const fixed32Type = 5;
+
+  proc unsignedVarintAppend(val:uint, ch: channel(true,iokind.little,false)) throws {
```

Comment:
It might be nice to use type aliases for this channel type, at the top of the file, like
```
type writingChannel = channel(true,iokind.little,false);
type readingChannel = channel(false,iokind.little,false);
```
this will reduce future maintenance work if these types change for some reason.

Reply:
Yes! Agreed, will do that.